app/src/androidTest/java/ch/epfl/sdp/blindly/match/MatchActivityTest.kt

```diff
@@ -0,0 +1,151 @@
+        onView(withId(R.id.card_stack_view)).check(
+            ViewAssertions.matches(
+                hasDescendant(
+                    withText(profiles[0].age.toString()))))
+    }
+
+    @Test
+    fun skipButtonDisplaySecondCard() {
+        launchScenario()
+
+        onView(withId(R.id.skip_button)).perform(click())
+        Thread.sleep(SLEEP_TIME)
+        onView(withId(R.id.card_stack_view)).check(
+            ViewAssertions.matches(
+                hasDescendant(
+                    withText(profiles[1].name))))
+        onView(withId(R.id.card_stack_view)).check(
+            ViewAssertions.matches(
+                hasDescendant(
+                    withText(profiles[1].age.toString()))))
+    }
+
+    @Test
+    fun likeButtonDisplaySecondCard() {
+        launchScenario()
+
+        onView(withId(R.id.like_button)).perform(click())
+        Thread.sleep(SLEEP_TIME)
+        onView(withId(R.id.card_stack_view)).check(
+            ViewAssertions.matches(
+                hasDescendant(
+                    withText(profiles[1].name))))
+        onView(withId(R.id.card_stack_view)).check(
+            ViewAssertions.matches(
+                hasDescendant(
+                    withText(profiles[1].age.toString()))))
+    }
+
+    @Test
+    fun rewindButtonHasNoEffectOnFirstCard() {
+        launchScenario()
+
+        onView(withId(R.id.rewind_button)).perform(click())
+        Thread.sleep(SLEEP_TIME)
+        onView(withId(R.id.card_stack_view)).check(
+            ViewAssertions.matches(
+                hasDescendant(
+                    withText(profiles[0].name))))
+        onView(withId(R.id.card_stack_view)).check(
+            ViewAssertions.matches(
+                hasDescendant(
+                    withText(profiles[0].age.toString()))))
+    }
+
+    @Test
+    fun rewindOnSecondCardReturnsToFirstCard() {
+        launchScenario()
+        onView(withId(R.id.like_button)).perform(click())
+        Thread.sleep(SLEEP_TIME)
+        onView(withId(R.id.rewind_button)).perform(click())
+        Thread.sleep(SLEEP_TIME)
+        onView(withId(R.id.card_stack_view)).check(
+            ViewAssertions.matches(
+                hasDescendant(
+                    withText(profiles[0].name))))
+        onView(withId(R.id.card_stack_view)).check(
+            ViewAssertions.matches(
+                hasDescendant(
+                    withText(profiles[0].age.toString()))))
+    }
+
+    @Test
+    fun swipeWorksAsButtons() {
+        launchScenario()
+
+        onView(withId(R.id.card_stack_view)).perform(swipeLeft())
+        Thread.sleep(SLEEP_TIME)
+        onView(withId(R.id.card_stack_view)).check(
+            ViewAssertions.matches(
+                hasDescendant(
+                    withText(profiles[1].name))))
+        onView(withId(R.id.card_stack_view)).check(
+            ViewAssertions.matches(
+                hasDescendant(
+                    withText(profiles[1].age.toString()))))
+    }
```

Comment:
Same comment. I feel like this test would pass anyway if you create the profiles before calling it.